### PR TITLE
US130517 [AQ] Settings Page Form Validation

### PR DESCRIPTION
**Pull request by Vitalii-Misechko**
[US130517](https://rally1.rallydev.com/#/?detail=/userstory/605755048093&fdp=true): [AQ] Settings Page Form Validation

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
  * [x] RTL
  * [x] Responsive
  * [x] Test cases
    * When a setting is off then its threshold gets disabled. In this case threshold's value does not affect validation.
    * When a setting is on then its threshold gets enabled and validation takes place.  
    * When a threshold gets validated its value must be in the range [0, 1]. Otherwise, the threshold is marked as invalid by the red icon
    * When Reliability Coefficient is on then Flag threshold must be populated.
    * When Quiz Reliability is on then High threshold must be populated. In addition, Reliability Coefficient gets toggled on and disabled. On top of that, Flag threshold must be less then High threshold. Otherwise, threshold is marked as invalid by red icon.
    * Toggling on Quiz Reliability should set valid validation state for Flag threshold (valid if it's populated, invalid if it greater than High threshold or out of range [0,1])
    * Toggling on Quiz Reliability should set valid validation state for High threshold (valid if it's populated, invalid if it less than Flag threshold or out of range [0, 1])
    * Updating the other threshold can remove the invalid icon from both inputs
    * Inline toast should repeat tooltip of invalid input
  * [x] Unit tests
* Security, devops, perf: N/A
* UX and docs
  * [x] Demo to Kevin
  * [x] Decision on how to show inline toast alert
  * [x] copy Better Error Alert Message
  
FYI @Brightspace/draco-vis

### CODE REVIEW

File: components/settings/quiz-details-settings-list.js [hunk 1]
```diff
@@ -250,6 +303,25 @@ class QuizDetailsSettingsList extends RtlMixin(Localizer(LitElement)) {
 		this.quizReliabilityHighThreshold = event.target.value;
 	}
 
+	_validateQuizReliabilityThresholds(event) {
+		const threshold = event.detail.forElement.value;
+		const pairThreshold = this._getPairThreshold(event.detail.forElement).value;
+		const isValid = (this.quizReliabilityFlagThreshold < this.quizReliabilityHighThreshold)
+			// this line ensure that inpute becomes valid if a pair input is empty (line above is invalid)
+			|| (threshold !== undefined && pairThreshold === undefined && threshold >= 0 && threshold <= 1);
+		event.detail.resolve(isValid);
```
Comment (NicholasHallman) [inline components/settings/quiz-details-settings-list.js:312]: Cool!

**Review by NicholasHallman — APPROVED** [review]
LGMT!

**Comment by Vitalii-Misechko** [comment]
> Is this new golden expected? Seems like an error image

Yes, it is. 404 is not a valid page in the frame.

**Review by bemailloux — APPROVED** [review]
LGTM! (although I guess you're still working out some details with Kevin?)

**Comment by Vitalii-Misechko** [comment]
> LGTM! (although I guess you're still working out some details with Kevin?)

Yes, I've found a UX case that Kevin wanted to avoid. Therefore, the PR can be changed

**Comment by Vitalii-Misechko** [comment]
Any objection I merge this into master? Last change to review this complex logic that is described in details of this PR )

**Review by bemailloux — APPROVED** [review]
LGTM! It looks like you have captured all cases you described here: https://d2l.slack.com/archives/CMNGJ22G0/p1632897930176700?thread_ts=1632897729.176400&cid=CMNGJ22G0

**Comment by Vitalii-Misechko** [comment]
> LGTM! It looks like you have captured all cases you described here: https://d2l.slack.com/archives/CMNGJ22G0/p1632897930176700?thread_ts=1632897729.176400&cid=CMNGJ22G0

Yep, that's right. I've described these cases in more detail in Test cases section of Quality notes

**Pull request merged**